src/runtime-rs/crates/service/src/event.rs

```diff
@@ -38,12 +38,12 @@ pub(crate) trait Forwarder {
 /// `TTRPC_ADDRESS` existing. Otherwise, fall back to `LogForwarder`.
 pub(crate) async fn new_event_publisher(namespace: &str) -> Result<Box<dyn Forwarder>> {
     let fwd: Box<dyn Forwarder> = match env::var(TTRPC_ADDRESS_ENV) {
-        Ok(address) => Box::new(
+        Ok(address) if !address.is_empty() => Box::new(
             ContainerdForwarder::new(namespace, &address)
                 .await
                 .context("new containerd forwarder")?,
         ),
-        Err(_) => Box::new(
+        Ok(_) | Err(_) => Box::new(
```

Comment:
maybe some comment here would be better to tell why do you take `Ok(_)` here after checking `Ok(address)`